lib/saml20.js

```diff
@@ -22,6 +24,33 @@ var algorithms = {
   }
 };
 
+function getAttributeType(value){
+  switch(typeof value) {
+    case "string":
+      return 'xs:string';
+    case "boolean":
+      return 'xs:boolean';
+    case "number":
+      return 'xs:long';
+    default:
+      return 'xs:anyType';
+  }
+}
+
+function getNameFormat(name){
+  if (is_uri(name)){
+    return 'urn:oasis:names:tc:SAML:2.0:attrname-format:uri';
+  }
+
+  //  Check that the name is a valid xs:Name -> https://www.w3.org/TR/xmlschema-2/#Name
+  if (xmlNameValidator.name(name).success){
+    return 'urn:oasis:names:tc:SAML:2.0:attrname-format:basic';
```

Comment:
include a comment about this value. what's is doing `xmlNameValidator.name(name).success` ?